sqlmesh/core/engine_adapter/base.py

```diff
@@ -407,6 +408,20 @@ def create_index(
         )
         self.execute(expression)
 
+    class _KeyForType(Enum):
+        TABLE = "TABLE_TYPE"
+        VIEW = "VIEW_TYPE"
```

Comment:
(Also discussed internally, sharing here for context)

I don't think we should associate the database object's type with the `*_properties` keys. For example, the physical layer can also contain views, i.e. if the model kind is `VIEW`, so setting `table_type` will be inconsistent.

Let's use a single common key within `physical_properties` and `virtual_properties`, e.g. `creatable_type = TEMPORARY`.

Reply:
I think this is done.